kitchen-sink-demo/src/app/demos/leaky-faucet/components/leaky-faucet.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'leaky-faucet',
+  template: `
+  <div>
+  <p>{{ status }}</p>
+  <button type="button" class="btn btn-danger"
+    (click)="toggleLeak()">
+    {{ start }} leaking ( calls a recursive function using setTimeout)
+  </button>
+  <div class="drip">Faucet: {{ drip }}</div>
+</div>
+  `,
+  styleUrls: ['./leaky-faucet.component.css']
+})
+
+
+export class LeakyFaucetComponent {
+
+  status = 'not leaking';
+  drip = 'start the drip';
+  start = 'Start';
```

Comment:
Variable name `start` doesn't make much sense since it can change to `stop`. Perhaps this is the `state`?